lib/providers/autoCompleteHelper.js

```diff
@@ -182,208 +180,42 @@ export default {
-		const sdk = Appc.selectedSdk();
-		const titaniumAPIPath = path.join(sdk.path, 'api.jsca');
-		const api = JSON.parse(fs.readFileSync(titaniumAPIPath, 'utf8'));
-
-		// generate tag list
-		const fns = fs.readdirSync(path.join(alloyPath, '/Alloy/commands/compile/parsers'));
-		let tagDic = {};
-		for (const fn of fns) {
-			if (!fn.endsWith('.js')) {
-				continue;
-			}
-			const ar = fn.split('.');
-			const tagName = ar[ar.length - 2];
-			if (tagName.indexOf('_') !== 0 && tagName[0] === tagName[0].toUpperCase()) {
-				tagDic[tagName] = {
-					apiName: fn.replace('.js', '')
-				};
-			} else if (tagName === '_ProxyProperty' && fn.indexOf('Ti.UI') === 0) {
-				tagDic[ar[ar.length - 3]] = { // Ti.UI.Window._ProxyProperty
-					apiName: fn.replace('.js', '').replace('._ProxyProperty', '')
-				};
-			}
-		}
-
-		// add missing tags
-		Object.assign(tagDic, {
-			View: {
-				apiName: 'Ti.UI.View'
-			},
-			Templates: {},
-			HeaderView: {},
-			FooterView: {},
-			ScrollView: {
-				apiName: 'Ti.UI.ScrollView'
-			},
-			Slider: {
-				apiName: 'Ti.UI.Slider'
-			},
-			TableViewRow: {
-				apiName: 'Ti.UI.TableViewRow'
-			},
-			Alloy: {},
-			ActivityIndicator: {
-				apiName: 'Ti.UI.ActivityIndicator'
-			},
-			WebView: {
-				apiName: 'Ti.UI.WebView'
-			}
-		});
-
-		// property list
-		const types = {};
-		const props = {};
-		api.types.forEach((type) => {
-			if (type.deprecated) {
-				return;
-			}
-
-			let propertyNamesOfType = [];
-			type.properties.forEach((prop) => {
-				if (prop.permission !== 'read-only' && prop.name.indexOf('Modules.') !== 0) {
-
-					propertyNamesOfType.push(prop.name);
-
-					if (prop.deprecated) {
-						prop.name += '|deprecated';
-					}
-
-					// property name
-					if (props[prop.name]) { // if duplicated property name - merge available values
-						Object.assign(props[prop.name], {
-							description: props[prop.name].description === prop.description.replace(/<p>|<\/p>/g, '') ? props[prop.name].description : ''
-						});
-						if (prop.constants.length) {
-							const values = props[prop.name].values ? props[prop.name].values.concat(prop.constants) : prop.constants;
-							props[prop.name].values = [ ...new Set(values) ];
-						}
-
-					} else {
-						props[prop.name] = {
-							description: prop.description.replace(/<p>|<\/p>/g, ''),
-							type: prop.type
-						};
-						if (prop.constants.length) {
-							props[prop.name].values = prop.constants;
-						}
-					}
+	async generateAutoCompleteSuggestions() {
+		const force = atom.config.get('appcelerator-titanium.general.generateAutoCompleteSuggestions');
+		const selectedSdk = Appc.selectedSdk();
+		try {
+			// Generate the completions
+			const [ alloy, sdk ] = await Promise.all([
+				completion.generateAlloyCompletions(force),
+				completion.generateSDKCompletions(force, selectedSdk.fullversion, selectedSdk.path)
+			]);
+			if (sdk || alloy) {
+				let message = 'Autocomplete suggestions generated for';
```

Comment:
<!--
  0 failure: 
  1 warning:  lib/providers/aut...
  
  
  DangerID: danger-id-Danger;
  File: lib/providers/autoCompleteHelper.js;
  Line: 193;
-->

- :warning: lib/providers/autoCompleteHelper.js line 193 – 'message' is assigned a value but never used. (no-unused-vars)